dokka-subprojects/analysis-kotlin-api/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/KotlinAnalysisPlugin.kt

```diff
@@ -29,6 +30,16 @@ public class KotlinAnalysisPlugin : DokkaPlugin() {
      */
     public val externalDocumentableProvider: ExtensionPoint<ExternalDocumentableProvider> by extensionPoint()
 
+    /**
+     * An extensions can rewrite Kotlin sample functions that come from the `@sample` KDoc tag.
+     * For example, it could be convenient to transform unit tests to samples.
+     * Dokka supports no more than one rewriter. By default, Dokka provides no rewriter.
```

Comment:
> Only because the same function can be transformed by multiple rewriters? 

Yes, it is enough. There are other corner cases as well. For example, should the result of a previous transforming be transformed again? `a()` -> `b()` -> `c()`. 
Anyway, some real use cases are needed to support it.